lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -2043,68 +2043,66 @@ quitStakePool ctx (ApiT wid) body = do
 
     genChange = delegationAddress @n
 
--- More testable helper for `listStakeKeys`.
+-- | More testable helper for `listStakeKeys`.
 listStakeKeys'
-    :: forall (n :: NetworkDiscriminant) m. Monad m
-    => UTxO.UTxO
-        -- ^ The wallet's UTxO
-    -> (Address -> Maybe RewardAccount)
+    :: forall (n :: NetworkDiscriminant)
+     . (Address -> Maybe RewardAccount)
         -- ^ Lookup reward account of addr
-    -> (Set RewardAccount -> m (Map RewardAccount (Maybe Coin)))
-        -- ^ Batch fetch of rewards
```

Comment:
While it is slightly more verbose than `[(RewardAccount, Coin)]`, it does move responsibility of "which accounts should we fetch rewards for" to the caller. I wanted to avoid this, as `listStakeKeys'` might discover new "foreign" stake keys which might as well fetch rewards for.

Reply:
I suggest to start simple, then add complexity later as needed.
It will make things easier for both yourself and the people who have to review the work.
If a monadic callback will be needed in the future to support foreign stake keys, then add it future.
Until then, make it as simple as possible - Value parameters instead of monadic function parameters.

Reply:
@rvl thanks a lot for having a look at this. I made some attempts to improve the in-code comments.

To my eyes, the monadic callback is not something needed for the future. Foreign stake keys in the UTxO with rewards could exist at this very moment (though always unlikely). And they are not known until the `listStakeKeys'` helper calculate them.

I also think it makes some conceptual sense to separate _how to fetch_ from _what to fetch_.

What I said in DM was that _ourKeys_ is currently just this:
```haskell
          let ourKeys = case mourAccount of
                   Just acc -> [(acc, 0, ourApiDelegation)]
                   Nothing -> []
```
and will have more entries in the future. 

That said, I want to have a closer look at https://github.com/input-output-hk/cardano-wallet/pull/2684#discussion_r643069025, which might affect things slightly, (in particular the need of the `lookupUsing`/ `Set -> Map` indirection), if we want to guarantee up-to-date account reward balances.

Reply:
@rvl on the topic of making things as simple as possible "until then" — I switched the function to actually use an un-cached, but batched `fetchAccountBalances`, which I think makes things look more logical, and further justifies the use of `(Set RewardAccount -> m (Map RewardAccount (Maybe Coin)))`:

<img width="1223" alt="Skärmavbild 2021-06-03 kl  15 58 30" src="https://user-images.githubusercontent.com/304423/120657291-a6e9cf00-c484-11eb-940d-8b93c0874844.png">